### ISSUE TITLE
Yarn

### ISSUE THREAD

**Issue by dmfenton** [asini/asini#56]
Have you looked into Yarn? I'm wondering if using Yarn would be more effective and easier to maintain than hoisting dependencies.

**Comment by gigabo** [comment]
Yes!  I actually have a branch with some work toward pluggable npm clients where I've done some testing with Yarn.  It's fast!  Here's a [performance breakdown](https://github.com/asini/asini/pull/36#issuecomment-253678684).  Basically: Yarn and hoisting both provide a big speed boost, and the improvements can be combined for very fast bootstrap!

```
              hoist

            no    yes
          +-----+-----+
        no| 30s | 12s |
yarn      |-----+-----+
       yes| 17s |  7s |
          +-----+-----+
```

**Comment by dmfenton** [comment]
Glad to hear it. Reading over issues posted on Lerna and Yarn, it looks like Yarn will not work unless all the packages are that listed as dependencies are on NPM. There's also issues with trying to run Yarn concurrently.

**Comment by gigabo** [comment]
Yep, I had to run with `--concurrency=1` with yarn.  Even then it's faster!

Plus, with hoisting, there's really only one main install to do anyway. ;)
